File: backend-java/src/main/java/com/citypilot/parking/services/OwnerService.java
package com.citypilot.parking.services;

import com.citypilot.parking.repositories.OwnerRepository;

import java.util.Map;

import static com.citypilot.parking.utils.JsonUtil.object;

public class OwnerService {
    private final OwnerRepository ownerRepository;

    public OwnerService(OwnerRepository ownerRepository) {
        this.ownerRepository = ownerRepository;
    }

    public Map<String, Object> getOwnerProfile() {
        return ownerRepository.getOwnerProfile();
    }

    public Map<String, Object> setRentStatus(Map<String, Object> input) {
        Object renting = input.get("renting");
        if (!(renting instanceof Boolean)) throw new IllegalArgumentException("renting must be boolean");
        return ownerRepository.updateRentStatus((Boolean) renting);
    }

    public Map<String, Object> createOwnerSpot(Map<String, Object> input) {
        Map<String, Object> spot = ownerRepository.createOwnerSpot(input);
        return object("spot", spot, "owner", ownerRepository.getOwnerProfile());
    }
}

File: backend/services/owner-service.js
const ownerRepository = require("../repositories/owner-repository");
const { badRequest } = require("../utils/errors");

function getOwnerProfile() {
  return ownerRepository.getOwnerProfile();
}

function setRentStatus(input) {
  if (typeof input.renting !== "boolean") {
    throw badRequest("renting must be boolean");
  }
  return ownerRepository.updateRentStatus(input.renting);
}

function createOwnerSpot(input) {
  const spot = ownerRepository.createOwnerSpot(input);
  return {
    spot,
    owner: ownerRepository.getOwnerProfile()
  };
}

module.exports = {
  getOwnerProfile,
  setRentStatus,
  createOwnerSpot
};
